share/picongpu/examples/LandauDamping/include/picongpu/param/fieldBackground.param

```diff
@@ -0,0 +1,135 @@
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file fieldBackground.param
+ *
+ * Load external background fields
+ */
+
+#pragma once
+
+
+namespace picongpu
+{
+    class FieldBackgroundE
+    {
+    public:
+        /* Add this additional field for pushing particles */
+        static constexpr bool InfluenceParticlePusher = true;
+
+        /* We use this to calculate your SI input back to our unit system */
+        PMACC_ALIGN(m_unitField, const float3_64);
+
+        HDINLINE FieldBackgroundE( const float3_64 unitField ) : m_unitField(unitField)
+        {}
+
+        /** Specify your background field E(r,t) here
+         *
+         * \param cellIdx The total cell id counted from the start at t = 0
+         * \param currentStep The current time step */
+        HDINLINE float3_X
+        operator()( const DataSpace<simDim>& cellIdx,
+                    const uint32_t currentStep ) const
+        {
+            /* calculate cells -> SI [m] */
+            float_64 const x_SI = cellIdx.x() * SI::CELL_WIDTH_SI;
+
+            /* note: you can also transform the time step to seconds by
+             *       multiplying with DELTA_T_SI */
+
+            /* specify your E-Field in V/m and convert to PIConGPU units */
+            float_X const arg = precisionCast< float_X >( x_SI * k_0 );
+            float_X const eps = 0.05;
+            float_X const C = SI::EPS0_SI * SI::BASE_CHARGE_SI;
+
+            return float3_X(
+                -C * eps * math::sin( arg ) / k_0 / m_unitField[0],
+                0.0,
+                0.0
+            );
+        }
+    };
+
+    class FieldBackgroundB
+    {
+    public:
+        /* Add this additional field for pushing particles */
+        static constexpr bool InfluenceParticlePusher = false;
+
+        /* We use this to calculate your SI input back to our unit system */
+        PMACC_ALIGN(m_unitField, const float3_64);
+
+        HDINLINE FieldBackgroundB( const float3_64 unitField ) : m_unitField(unitField)
+        {}
+
+        /** Specify your background field B(r,t) here
+         *
+         * \param cellIdx The total cell id counted from the start at t=0
+         * \param currentStep The current time step */
+        HDINLINE float3_X
+        operator()( const DataSpace<simDim>& cellIdx,
+                    const uint32_t currentStep ) const
+        {
+            /* example: periodicity of 20 microns ( = 2.0e-5 m) */
+            constexpr float_64 period_SI(20.0e-6);
+            /* calculate cells -> SI -> [m] */
+            const float_64 y_SI = cellIdx.y() * SI::CELL_HEIGHT_SI;
+            /* note: you can also transform the time step to seconds by
+             *       multiplying with DELTA_T_SI */
+
+            /* specify your B-Field in T and convert to PIConGPU units */
+            const float_X sinArg = precisionCast<float_X>( y_SI / period_SI * 2.0 * PI );
+            return float3_X(0.0, math::cos( sinArg ) / m_unitField[1], 0.0);
```

Comment:
note: this mix 32bit and 64bit operation. This is not critical because the 32bit value will be implicit cast to 64bit but please check if this is what you want.

Reply:
this is taken from the default and needs adjustment